http-client: fix setting response headers

HustonMmmavr opened the pull request:
* Fix behaviour of `curl_easy_header_cb` (from src/http.c).
  By default CURLOPT_HEADERFUNCTION collects headers of all
  responses received after initiating a request and not just
  the final response. But in case of redirection, only headers
  from the final response is needed.

Close #6101

ligurio commented:
(PR is in draft status, but I'll anyway keep feedback about proposed changes.)

Alexander, thanks for the changes!

There are two typos:
- commit "    http-client: fix setting response headers" has a wrong path to file ('src/http.c' -> 'src/httpc.c').
- "verify headers after redirecrt is correct" -> "verify headers after redirect is correct"
- why do we need to remove a line?
```c
@@ -168,7 +190,6 @@ httpc_request_delete(struct httpc_request *req)
 {
        if (req->headers != NULL)
                curl_slist_free_all(req->headers);
-
        curl_request_destroy(&req->curl_request);
 
        ibuf_destroy(&req->body);
```
- why we need to set follow_location using C API and directly?
```c
@@ -149,6 +170,7 @@ httpc_request_new(struct httpc_env *env, const char *method,
        curl_easy_setopt(req->curl_request.easy, CURLOPT_URL, url);
 
        curl_easy_setopt(req->curl_request.easy, CURLOPT_FOLLOWLOCATION, 1);
+       req->follow_location = true;
        curl_easy_setopt(req->curl_request.easy, CURLOPT_SSL_VERIFYPEER, 1);
        curl_easy_setopt(req->curl_request.easy, CURLOPT_WRITEFUNCTION,
                         curl_easy_write_cb);
```

HustonMmmavr commented:
Hello! Thank you for the review!

I’ve fixed commit message and commentary, restored a line.
Also I've opened a new [PR](https://github.com/tarantool/tarantool/pull/6324)

About setting follow_location directly: CURLOPT_FOLLOWLOCATION
Was set by default to 1 (true) at previous patches, I’ve failed to find curl methods to get this CURLOPT_FOLLOWLOCATION from the request structure, but this flag is needed to determine request pipeline (was it redirected or not), so I’ve added additional field to tarantool srtucture and set it by default to true

ligurio commented:
Seems this PR can be closed as you are open a new one (https://github.com/tarantool/tarantool/pull/6324) @HustonMmmavr

ligurio commented:
PR replaced with https://github.com/tarantool/tarantool/pull/6324
